FirebaseCore/CHANGELOG.md

```diff
@@ -1,3 +1,6 @@
+# Firebase 10.8.0
+- [changed] The Firebase Swift package now requires the Swift 5.6 toolchain (Xcode 13.3) to build.
```

Comment:
I don't fully grasp the implications of changing `swift-tools-version`. My understanding is that raising it can "unlock" new API that can be used in the Package.swift. And from [here](https://developer.apple.com/documentation/packagedescription/package#:~:text=tools%2Dversion%3A%205.6-,The%20Swift%20tools%20version,-declares%20the%20version), that it sets the minimum tools version required to use the package. So, I'm assuming that if I was using the Swift 5.5 toolchain or under, I could no longer build Firebase?

What other existing use cases break when this is raised?

Reply:
That's the only use case that I'm aware of that would break by increasing the `swift-tools-version`. The goal of this change is to break with a more descriptive error message when attempting to use the Firebase package with Xcode versions that we don't support.

E.g., if a developer uses Xcode 13.2 (which includes the Swift 5.5 toolchain -- see [swiftversion.net](https://swiftversion.net/)) they'll run into various [compilation errors](https://swiftpackageindex.com/builds/B385F8AC-D578-426E-8502-2B996974B1AC):
```
StorageDeleteTask.swift:74:46: warning: conditional downcast from 'Error?' to 'NSError' is a bridging
conversion; did you mean to use 'as'?
        self?.fetcherCompletion?(data, error as? NSError)
                                       ~~~~~ ^~~ ~~~~~~~
                                             as         ?
```
[or](https://swiftpackageindex.com/builds/14CE51EE-9312-4550-AC84-26E5A0AE7084):
```
SessionStartEvent.swift:192:56: error: cannot convert value of type 'UnsafeRawPointer' to
expected argument type 'UnsafePointer<pb_byte_t>?' (aka 'Optional<UnsafePointer<UInt8>>')
    var istream: pb_istream_t = pb_istream_from_buffer(bytes, transportBytes.count)
```

These don't appear on CI because we're using Xcode 13.3.1 (Swift 5.6) for our testing.  Setting `swift-tools-version:5.6` results in the error `package at 'some-path/firebase-ios-sdk' is using Swift tools version 5.6.0 but the installed version is 5.5.0` when opening the package in Xcode 13.2.1. IMHO, this is more likely to prompt the developer to upgrade Xcode and/or easier to search for.

Reply:
Thanks @andrewheard–– makes sense!

> How about something like "The Firebase Swift Package Manager distribution now requires the Swift 5.6 toolchain (Xcode 13.3) to build. This further enforces the official Firebase requirement for Xcode 13.3.1."

I second this.

Reply:
And thanks @paulb777 for the changelog text! Copied exactly.